[ANGLE] Update ANGLE to 2023-01-30 (786d237f39f1ea0d85781c039d4218fcd2679156)

djg opened the pull request:
#### a30b362a22515764d1dfa2fe4018fb716d11a702
<pre>
[ANGLE] Update ANGLE to 2023-01-30 (786d237f39f1ea0d85781c039d4218fcd2679156)
<a href="https://bugs.webkit.org/show_bug.cgi?id=251415">https://bugs.webkit.org/show_bug.cgi?id=251415</a>
rdar://problem/104850019

Reviewed by Kimmo Kinnunen.

Update ANGLE to 2023-01-30 (786d237f39f1ea0d85781c039d4218fcd2679156)

Contains upstream commits:
git log --oneline e5671e16b83ff26b25c2a831ff9cdc3930f8c06d..786d237f39f1ea0d85781c039d4218fcd2679156 --pretty=%h %s
786d237f3 Vulkan: Fixed bug in rx::vk::SharedFence with repeated init().
dbd47e378 Revert &quot;Metal: Avoid leaking buffers for GPU access for non-discrete&quot;
dd8021d98 Added &quot;git worktree&quot; support into &quot;commit_id.py&quot;.
73cc578b4 Disable mutable texture upload due to render issue
b8bfe6f6c Metal: rewrite default uniforms and uniform blocks
53b659501 Vulkan: Fold ThreadSafeCommandProcessor back to CommandProcessor
6fb2da906 Do not use std::locale in Vulkan memory allocation tracker
0cdecfad0 Vulkan: Merge ThreadSafeCommandQueue back into CommandQueue
7101b3747 Vulkan: Enable AndroidNativeFence_ClientWait test
2e685d6cd Roll VK-GL-CTS from 821e153ac85a to bbf05746bf87 (8 revisions)
e8afed5b9 Roll Chromium from fb367de4643f to 97cd4b79248f (792 revisions)
69eab34b0 Vulkan: Fixed UB creating VkPipeline in warmUpPipelineCache().
4b72bd1b8 Roll vulkan-deps from d723200bd4b5 to be674ea25e15 (14 revisions)
0cde45b7c Metal: Lazily grab the shader lib for provoking vertex helper
e1d5d8c08 Include stdint.h instead of cstdint in platform.h
369b320f9 Metal: Avoid leaking buffers for GPU access for non-discrete
4716d85e1 Tests: Add LIMBO trace
b96d19559 Implement GL_ARM_shader_framebuffer_fetch
60ad92e43 Tests: Add Goddess Of Victory: Nikke trace
54f446394 Vulkan: Make RendererVk only have one CommandQueue object
06720c9bc Add binding to DescriptorInfoDesc.
0931b6799 FrameCapture: Update shader capture for CRLF
a8146c5d0 Roll vulkan-deps from 8a9f79f08d79 to d723200bd4b5 (1 revision)
3d182016f Roll Chromium from 5c03f547ddf9 to fb367de4643f (549 revisions)
83150e249 Roll vulkan-deps from a2f348c2039f to 8a9f79f08d79 (22 revisions)
1d9b8d802 Reland &quot;Add support for glShaderBinary&quot;
17ac69431 Move d3d11 BCX textures to the &quot;UNREACHABLE&quot; section for clears
6938b883f Tests: Add Dragon Ball Z Dokkan Battle trace
081764c84 Remove checks for A8 devices
6eb6cb2ed Declare size_t in Constants.h by including stddef.h
128eb01dd Vulkan: Only call checkCompletedCommands if needed.
bed19c471 Tests: Add Bubble Shooter and Friends trace
226fd6667 Vulkan: Make CommandProcessor::checkCompletedCommands immediate check
63d9675f0 Add documentation for enabling Metal validation layers.
ef0fe6388 Implement EXT_polygon_offset_clamp
9edc4686a Vulkan: Fix SharedFence::mRefCountedFence leak
0133b6f18 Add GL_ARM_shader_framebuffer_fetch builtins
18657894b Add features to dump and replace shader source.
862e3e95c Skip failing capture/replay multithreaded test
dfcdae3bf FrameCapture: Add blend state updates to resetCalls.
0c9cbf39f Revert &quot;Add support for glShaderBinary&quot;
5e43562b5 Roll Chromium from 701fadbce737 to 5c03f547ddf9 (614 revisions)
b64b573ba Capture/Replay: disable more multi-threaded tests.
00b94fdae Vulkan: CommandProcessor::WaitForWorkComplete only if needed
1e3197f47 Metal: Remove the UNIMPLEMENTED from setBinaryRetrievableHint
228973e73 Add support for glShaderBinary
1d9984bd4 Vulkan: Make CommandProcessor use ThreadSafeCommandQueue
6a5e61cd2 Vulkan: Make ThreadSafeCommandQueue&apos;s wait without lock
c9094ca6b Skip flaky tests dEQP-GLES31.functional.image_load_store.buffer*
11608a8b7 Adjust STENCIL_INDEX8 caps
f26cfec91 Roll Chromium from 7d74066a9406 to 701fadbce737 (593 revisions)
660e4a7c0 Don&apos;t override D16_UNORM to D32_FLOAT on Apple GPUs.

Canonical link: <a href="https://commits.webkit.org/259617@main">https://commits.webkit.org/259617@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f46100a71c5bc9f72bd8eb9ae6f8daf90457dcc0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105411 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38283 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114666 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/174829 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15672 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5417 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97715 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114545 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111169 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12108 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95089 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39590 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93974 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26724 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7790 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28081 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7912 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13933 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47628 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6633 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9704 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple reviewed (approved):
I think you amended Constants.h without updating the changes.diff.

djg committed:
[ANGLE] Update ANGLE to 2023-01-30 (786d237f39f1ea0d85781c039d4218fcd2679156)
https://bugs.webkit.org/show_bug.cgi?id=251415
rdar://problem/104850019

Reviewed by Kimmo Kinnunen.

Update ANGLE to 2023-01-30 (786d237f39f1ea0d85781c039d4218fcd2679156)

Contains upstream commits:
git log --oneline e5671e16b83ff26b25c2a831ff9cdc3930f8c06d..786d237f39f1ea0d85781c039d4218fcd2679156 --pretty=%h %s
786d237f3 Vulkan: Fixed bug in rx::vk::SharedFence with repeated init().
dbd47e378 Revert "Metal: Avoid leaking buffers for GPU access for non-discrete"
dd8021d98 Added "git worktree" support into "commit_id.py".
73cc578b4 Disable mutable texture upload due to render issue
b8bfe6f6c Metal: rewrite default uniforms and uniform blocks
53b659501 Vulkan: Fold ThreadSafeCommandProcessor back to CommandProcessor
6fb2da906 Do not use std::locale in Vulkan memory allocation tracker
0cdecfad0 Vulkan: Merge ThreadSafeCommandQueue back into CommandQueue
7101b3747 Vulkan: Enable AndroidNativeFence_ClientWait test
2e685d6cd Roll VK-GL-CTS from 821e153ac85a to bbf05746bf87 (8 revisions)
e8afed5b9 Roll Chromium from fb367de4643f to 97cd4b79248f (792 revisions)
69eab34b0 Vulkan: Fixed UB creating VkPipeline in warmUpPipelineCache().
4b72bd1b8 Roll vulkan-deps from d723200bd4b5 to be674ea25e15 (14 revisions)
0cde45b7c Metal: Lazily grab the shader lib for provoking vertex helper
e1d5d8c08 Include stdint.h instead of cstdint in platform.h
369b320f9 Metal: Avoid leaking buffers for GPU access for non-discrete
4716d85e1 Tests: Add LIMBO trace
b96d19559 Implement GL_ARM_shader_framebuffer_fetch
60ad92e43 Tests: Add Goddess Of Victory: Nikke trace
54f446394 Vulkan: Make RendererVk only have one CommandQueue object
06720c9bc Add binding to DescriptorInfoDesc.
0931b6799 FrameCapture: Update shader capture for CRLF
a8146c5d0 Roll vulkan-deps from 8a9f79f08d79 to d723200bd4b5 (1 revision)
3d182016f Roll Chromium from 5c03f547ddf9 to fb367de4643f (549 revisions)
83150e249 Roll vulkan-deps from a2f348c2039f to 8a9f79f08d79 (22 revisions)
1d9b8d802 Reland "Add support for glShaderBinary"
17ac69431 Move d3d11 BCX textures to the "UNREACHABLE" section for clears
6938b883f Tests: Add Dragon Ball Z Dokkan Battle trace
081764c84 Remove checks for A8 devices
6eb6cb2ed Declare size_t in Constants.h by including stddef.h
128eb01dd Vulkan: Only call checkCompletedCommands if needed.
bed19c471 Tests: Add Bubble Shooter and Friends trace
226fd6667 Vulkan: Make CommandProcessor::checkCompletedCommands immediate check
63d9675f0 Add documentation for enabling Metal validation layers.
ef0fe6388 Implement EXT_polygon_offset_clamp
9edc4686a Vulkan: Fix SharedFence::mRefCountedFence leak
0133b6f18 Add GL_ARM_shader_framebuffer_fetch builtins
18657894b Add features to dump and replace shader source.
862e3e95c Skip failing capture/replay multithreaded test
dfcdae3bf FrameCapture: Add blend state updates to resetCalls.
0c9cbf39f Revert "Add support for glShaderBinary"
5e43562b5 Roll Chromium from 701fadbce737 to 5c03f547ddf9 (614 revisions)
b64b573ba Capture/Replay: disable more multi-threaded tests.
00b94fdae Vulkan: CommandProcessor::WaitForWorkComplete only if needed
1e3197f47 Metal: Remove the UNIMPLEMENTED from setBinaryRetrievableHint
228973e73 Add support for glShaderBinary
1d9984bd4 Vulkan: Make CommandProcessor use ThreadSafeCommandQueue
6a5e61cd2 Vulkan: Make ThreadSafeCommandQueue's wait without lock
c9094ca6b Skip flaky tests dEQP-GLES31.functional.image_load_store.buffer*
11608a8b7 Adjust STENCIL_INDEX8 caps
f26cfec91 Roll Chromium from 7d74066a9406 to 701fadbce737 (593 revisions)
660e4a7c0 Don't override D16_UNORM to D32_FLOAT on Apple GPUs.

Canonical link: https://commits.webkit.org/259617@main
